src/lib.rs

```diff
@@ -77,27 +77,40 @@ use std::fmt;
 use std::error;
 use libc::PATH_MAX;
 
+/// Nix Result Type
 pub type Result<T> = result::Result<T, Error>;
 
+/// Nix Error Type
+///
+/// The nix error type provides a common way of dealing with
+/// various system system/libc calls that might fail.  Each
+/// error has a corresponding errno (usually the one from the
+/// underlying OS) to which it can be mapped in addition to
+/// implementing other common traits.
```

Comment:
should we doc `InvalidPath` or just punt on it in hopes we actually kill it soon?